presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -415,7 +442,7 @@ public PinotQueryGeneratorContext visitAggregation(AggregationNode node, PinotQu
             checkSupported(!forbidBrokerQueries, "Cannot push aggregation in segment mode");
 
             // 2nd pass
-            LinkedHashMap<VariableReferenceExpression, Selection> newSelections = new LinkedHashMap<>();
+            Map<VariableReferenceExpression, Selection> newSelections = new LinkedHashMap<>();
```

Comment:
Here the reason is that in our PQL generation test, we create AggregationPlanNode by setting the Aggregations directly, if we use HashMap then generatedPql cannot guarantee the ordering of aggregation functions. Otherwise we need to set a ProjectNode always outside an AggregationPlanNode

Reply:
Lets please fix the tests instead of complicating the production code like this. Even if it means more changes.

Reply:
will do :)

Reply:
The issue is that in visitAggregation method, I didn't explicitly generate outputs, and infer it when calling context.withAggregation by using outputs.addAll(newSelection.keySet()). This is fine when selection is an ordered map.
So the change I made is to generate outputs in visitAggregation method.